File: src/main/java/com/bdev/hogwarts_api/data/converter/group/GroupDtoConverter.java
package com.bdev.hogwarts_api.data.converter.group;

import com.bdev.hogwarts_api.data.dto.group.Group;
import com.bdev.hogwarts_api.data.dto.group.GroupLesson;
import com.bdev.hogwarts_api.data.model.group.GroupLessonModel;
import com.bdev.hogwarts_api.data.model.group.GroupModel;
import com.bdev.hogwarts_api.utils.EncodingUtils;

import java.util.stream.Collectors;

import static com.bdev.hogwarts_api.utils.EncodingUtils.toBase64;
import static java.util.stream.Collectors.toList;

public class GroupDtoConverter {
    public static GroupModel convert(Group group) {
        GroupModel groupModel = new GroupModel();

        groupModel.setId(group.getId());
        groupModel.setCabinetId(group.getCabinetId());
        groupModel.setName(toBase64(group.getName()));
        groupModel.setBookName(toBase64(group.getBookName()));
        groupModel.setLessons(group.getLessons().stream().map(it -> convertLesson(groupModel, it)).collect(toList()));
        groupModel.setAge(group.getAge());
        groupModel.setEducationLevel(group.getEducationLevel());
        groupModel.setColor(group.getColor());

        return groupModel;
    }

    private static GroupLessonModel convertLesson(GroupModel groupModel, GroupLesson lesson) {
        GroupLessonModel groupLessonModel = new GroupLessonModel();

        groupLessonModel.setId(lesson.getId());
        groupLessonModel.setGroup(groupModel);
        groupLessonModel.setTeacherId(lesson.getTeacherId());
        groupLessonModel.setDay(lesson.getDay());
        groupLessonModel.setStartTime(lesson.getStartTime());
        groupLessonModel.setFinishTime(lesson.getFinishTime());

        return groupLessonModel;
    }
}

File: src/main/java/com/bdev/hogwarts_api/data/converter/group/GroupDtoConverter.kt
package com.bdev.hogwarts_api.data.converter.group

import com.bdev.hogwarts_api.data.dto.group.Group
import com.bdev.hogwarts_api.data.dto.group.GroupLesson
import com.bdev.hogwarts_api.data.model.group.GroupLessonModel
import com.bdev.hogwarts_api.data.model.group.GroupModel
import com.bdev.hogwarts_api.utils.EncodingUtils.toBase64
import java.util.*

object GroupDtoConverter {
    fun convert(group: Group): GroupModel {
        val groupModel = GroupModel()

        groupModel.id = group.id
        groupModel.cabinetId = group.cabinetId
        groupModel.headTeacherLogin = group.headTeacherLogin
        groupModel.type = group.type
        groupModel.lessons = group.lessons.map { it -> convertLesson(groupModel, it) }.toMutableList()
        groupModel.age = group.age
        groupModel.educationLevel = group.educationLevel
        groupModel.color = group.color

        return groupModel
    }

    private fun convertLesson(groupModel: GroupModel, lesson: GroupLesson): GroupLessonModel {
        val groupLessonModel = GroupLessonModel()

        groupLessonModel.id = lesson.id
        groupLessonModel.group = groupModel
        groupLessonModel.teacherLogin = lesson.teacherLogin
        groupLessonModel.isOnline = lesson.isOnline
        groupLessonModel.ignoreSingleStudentPricing = lesson.ignoreSingleStudentPricing
        groupLessonModel.day = lesson.day
        groupLessonModel.startTime = lesson.startTime
        groupLessonModel.finishTime = lesson.finishTime
        groupLessonModel.creationTime = lesson.creationTime
        groupLessonModel.deactivationTime = lesson.deactivationTime

        return groupLessonModel
    }
}
